Did you validate smoketests? The immadiately fail for me:

```
DDCLIENT test - run #2 previous good/bad (0/1)
test_01_dyndns_service_standard (__main__.TestServiceDDNS.test_01_dyndns_service_standard) ... FAIL

======================================================================
FAIL: test_01_dyndns_service_standard (__main__.TestServiceDDNS.test_01_dyndns_service_standard)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/libexec/vyos/tests/smoke/cli/test_service_dns_dynamic.py", line 54, in tearDown
    self.assertFalse(process_named_running(DDCLIENT_PNAME))
AssertionError: 5232 is not false

----------------------------------------------------------------------
Ran 1 test in 23.499s

FAILED (failures=1)
```

Using:

```bash
#!/bin/sh

good=0
bad=0
for ii in $(seq 1 1000)
do
    echo "DDCLIENT test - run #$ii previous good/bad ($good/$bad)"
    /usr/libexec/vyos/tests/smoke/cli/test_service_dns_dynamic.py
    if [ "x$?" != "x0" ]; then
        bad=$((bad+1))
        continue
    fi
    good=$((good+1))
done
```

--- a/data/templates/dns-dynamic/ddclient.conf.j2
+++ b/data/templates/dns-dynamic/ddclient.conf.j2
@@ -24,7 +24,6 @@ if{{ ipv }}={{ address }}, \
 daemon={{ interval }}
 syslog=yes
 ssl=yes
-pid={{ config_file | replace('.conf', '.pid') }}
 cache={{ config_file | replace('.conf', '.cache') }}
 {# ddclient default (web=dyndns) doesn't support ssl and results in process lockup #}
 web=googledomains

--- a/data/templates/dns-dynamic/override.conf.j2
+++ b/data/templates/dns-dynamic/override.conf.j2
@@ -1,10 +1,11 @@
 {% set vrf_command = 'ip vrf exec ' ~ vrf ~ ' ' if vrf is vyos_defined else '' %}
 [Unit]
 ConditionPathExists={{ config_file }}
+Wants=
 After=vyos-router.service
 
 [Service]
-PIDFile={{ config_file | replace('.conf', '.pid') }}
 EnvironmentFile=
 ExecStart=
-ExecStart={{ vrf_command }}/usr/bin/ddclient -file {{ config_file }}
+ExecStart={{ vrf_command }}/usr/bin/ddclient --file {{ config_file }} --foreground
+Restart=always

--- a/smoketest/scripts/cli/test_service_dns_dynamic.py
+++ b/smoketest/scripts/cli/test_service_dns_dynamic.py
@@ -1,6 +1,6 @@
 #!/usr/bin/env python3
 #
-# Copyright (C) 2019-2024 VyOS maintainers and contributors
+# Copyright (C) 2019-2025 VyOS maintainers and contributors
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License version 2 or later as
@@ -38,9 +38,13 @@ ttl = '300'
 interface = 'eth0'
 
 class TestServiceDDNS(VyOSUnitTestSHIM.TestCase):
-    def setUp(self):
-        # Always start with a clean CLI instance
-        self.cli_delete(base_path)
+    @classmethod
+    def setUpClass(cls):
+        super(TestServiceDDNS, cls).setUpClass()
+
+        # ensure we can also run this test on a live system - so lets clean
+        # out the current configuration :)
+        cls.cli_delete(cls, base_path)
 
     def tearDown(self):
         # Check for running process
@@ -336,8 +340,8 @@ class TestServiceDDNS(VyOSUnitTestSHIM.TestCase):
 
         # Check for process in VRF
         systemd_override = cmd(f'cat {DDCLIENT_SYSTEMD_UNIT}')
-        self.assertIn(f'ExecStart=ip vrf exec {vrf_name} /usr/bin/ddclient -file {DDCLIENT_CONF}',
-                      systemd_override)
+        self.assertIn(f'ExecStart=ip vrf exec {vrf_name} /usr/bin/ddclient ' \
+                      f'--file {DDCLIENT_CONF} --foreground', systemd_override)
 
         # Check for process in VRF
         proc = cmd(f'ip vrf pids {vrf_name}')

